src/boosting/sample_strategy.cpp

```diff
@@ -0,0 +1,25 @@
+/*!
+ * Copyright (c) 2021 Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE file in the project root for license information.
+ */
+
+#include <LightGBM/sample_strategy.h>
+#include "goss.hpp"
+#include "bagging.hpp"
+
+namespace LightGBM {
+
+SampleStrategy* SampleStrategy::CreateSampleStrategy(const Config* config, const Dataset* train_data, const ObjectiveFunction* objective_function, int num_tree_per_iteration) {
+  bool use_goss_as_boosting = config->boosting == std::string("goss");
+  bool use_goss_as_strategy = config->data_sample_strategy == std::string("goss");
+  if (use_goss_as_boosting) {
+    Log::Warning("Found boosting_type=goss. For backwards compatibility reasons, LightGBM interprets this as boosting_type=gbdt, data_sample_strategy=goss. To suppress this warning, set data_sample_strategy=goss instead.");
```

Comment:
```suggestion
    Log::Warning("Found boosting=goss. For backwards compatibility reasons, LightGBM interprets this as boosting=gbdt, data_sample_strategy=goss. To suppress this warning, set data_sample_strategy=goss instead.");
```

`boosting_type` is an alias for `boosting`. I think that we should try to make this log message reference the "main" parameter instead of the alias. Otherwise, users might be confused that `boosting_type` is mentioned in this warning even if they didn't include it in parameters.

Reply:
I agree.